ibis/expr/types/relations.py

```diff
@@ -3263,7 +3255,13 @@ def sql(self, query: str, dialect: str | None = None) -> ir.Table:
         if dialect is not None:
             backend = self._find_backend()
             query = backend._transpile_sql(query, dialect=dialect)
-        op = ops.SQLStringView(child=self, name=next(_ALIASES), query=query)
+
+        child = (
+            self
+            if isinstance(self.op(), ops.View)
+            else self.alias(util.gen_name("sql_string_view"))
```

Comment:
What happens when we generate an alias? How is query going to use that name?

Reply:
There's no requirement that a subsequent query use this name, it's merely to satisfy the constraint I introduced that `SQLStringView`'s `child` member type is `View` and not an arbitrary relation.

Reply:
Maybe there's a better way to do that that avoids constructing a useless CTE.